lib/exercises.rb

```diff
@@ -2,19 +2,62 @@
 
 # This method will return an array of arrays.
 # Each subarray will have strings which are anagrams of each other
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(w * n^2) ?
+# THINKING....
+# Iterating through the string = O(n), summing each word = O(w); O(n * w)
+# checking for the word's sum in the hash and shovel onto array = O(1)
+# returning the hash values = O(n) b/c if no words are anagrams, our hash would be the
+# same size as the original array.
+
+# Space Complexity: O(n); The hash ends up being the same size as the string/input size.
 
 def grouped_anagrams(strings)
-  raise NotImplementedError, "Method hasn't been implemented yet!"
+  return strings if strings.empty?
+  
+  anagrams = {}
+  # for each word, if the sum is in the hash
+  strings.each do |word|
+    if anagrams[word.sum].nil?
+      anagrams[word.sum] = [word]
+    else
+      anagrams[word.sum] << word
+    end
+  end
```

Comment:
Looking at the BigO here.  Strings.each n, inside you do word.sum which is O(w), so this is O(nw).  The final return statement is O(n), but it's not in another loop so we can ignore it.